2018 Robot/src/org/usfirst/frc/team4999/robot/subsystems/Lift.java

```diff
@@ -50,53 +52,33 @@ public boolean isHighSpeed() {
 		return shifter.get() == DoubleSolenoid.Value.kForward;
 	}
 	public void shiftHigh() {
+		currentLiftPID = fastLiftPID;
 		shifter.set(DoubleSolenoid.Value.kForward);
 	}
 	public void shiftLow() {
+		currentLiftPID = slowLiftPID;
 		shifter.set(DoubleSolenoid.Value.kReverse);
 	}
 	
 	public void driveLiftPID() {
-		if(isHighSpeed())
-			driveFastLiftPID();
+		if(currentLiftPID.isEnabled() && !isBraked())
+			set(currentLiftPID.get());
 		else
-			driveSlowLiftPID();
+			set(0);
 	}
 	
 	public double getCurrentHeight( ) {
 		return encoder.getDistance();
 	}
 	
 	public boolean isOnTarget() {
-		if(isHighSpeed())
-			return fastLiftPID.onTargetForTime();
-		else
-			return slowLiftPID.onTargetForTime();
-	}
-	
-	public void driveSlowLiftPID() {
-		if(slowLiftPID.isEnabled() && !isBraked())
-			set(slowLiftPID.get());
-		else
-			set(0);
-	}
-	public void driveFastLiftPID() {
-		if(fastLiftPID.isEnabled() && !isBraked())
-			set(fastLiftPID.get());
-		else
-			set(0);
-	}
-	public void drivePID() {
-		if(isHighSpeed())
-			driveFastLiftPID();
-		else
-			driveSlowLiftPID();
+		return currentLiftPID.onTargetForTime();
 	}
 	
 	public void set(double power) {
 		// Anti-jerk: Clip the difference between the requested power and the current power
 		double delta = power - motors.get();
-		delta = clip(delta, -MAX_MOTOR_DELTA, MAX_MOTOR_DELTA);
+		delta = Utils.clip(delta, -MAX_MOTOR_DELTA, MAX_MOTOR_DELTA);
 		double c_power = motors.get() + delta;
```

Comment:
I'm still concerned about this.  Have you been able to test this behavior on last year's bot using the drive?  You may for that it doesn't do what you expect.